core-rust/mesh-api-server/src/mesh_api/handlers/mempool_transaction.rs

```diff
@@ -19,27 +19,60 @@ pub(crate) async fn handle_mempool_transaction(
     )
     .map_err(|err| err.into_response_error("intent_hash"))?;
 
-    if mempool
-        .get_mempool_payload_hashes_for_intent(&intent_hash)
-        .is_empty()
-    {
+    let payload_hashes = mempool.get_mempool_payload_hashes_for_intent(&intent_hash);
+    let notarized_transaction_hash = if payload_hashes.is_empty() {
         return Err(
             ResponseError::from(ApiError::TransactionNotFound).with_details(format!(
-                "transaction {} not found in mempool transactions",
+                "Transaction {} not found in mempool transactions",
                 &request.transaction_identifier.hash
             )),
         );
-    }
+    } else {
+        payload_hashes.get(0).unwrap()
+    };
+
+    let user_transaction = match mempool.get_mempool_payload(&notarized_transaction_hash) {
+        Some(transaction) => transaction.raw.into_typed().map_err(|_| {
```

Comment:
This shouldn't be possible, because it's known to be executable, therefore it must at least be parsable. So we should be able to unwrap here :)

Reply:
Done